pkg/controller/function/function_controller.go

```diff
@@ -83,62 +107,178 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 }
 
-var _ reconcile.Reconciler = &ReconcileFunction{}
+var (
+	// name of function config
+	fnConfigName = getEnvDefault("CONTROLLER_CONFIGMAP", "fn-config")
+
+	// namespace of function config
+	fnConfigNamespace = getEnvDefault("CONTROLLER_CONFIGMAP_NS", "default")
+
+	// name of build-template
+	buildTemplateName                      = getEnvDefault("BUILD_TEMPLATE", "function-kaniko")
+	_                 reconcile.Reconciler = &ReconcileFunction{}
 
+	// "build and push step"
+	buildAndPushStep = "build-step-build-and-push"
+)
+
+// ReconcileFunction is the controller.Reconciler implementation for Function objects
 // ReconcileFunction reconciles a Function object
 type ReconcileFunction struct {
 	client.Client
 	scheme *runtime.Scheme
 }
 
+func getEnvDefault(envName string, defaultValue string) string {
+	// use default value if environment variable is empty
+	var value string
+	if value = os.Getenv(envName); value == "" {
+		return defaultValue
+	}
+	return value
+}
+
 // Reconcile reads that state of the cluster for a Function object and makes changes based on the state read
 // and what is in the Function.Spec
+// +kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="runtime.kyma-project.io",resources=functions,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="runtime.kyma-project.io",resources=functions/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups="admissionregistration.k8s.io",resources=mutatingwebhookconfigurations;validatingwebhookconfigurations,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups="serving.knative.dev",resources=services,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups="",resources=serviceaccounts,verbs=get;list
-// +kubebuilder:rbac:groups=";apps;extensions",resources=deployments,verbs=create;get;delete;list;update;patch
+// +kubebuilder:rbac:groups="serving.knative.dev",resources=services;routes;configurations;revisions,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="build.knative.dev",resources=builds;buildtemplates;clusterbuildtemplates;services,verbs=get;list;create;update;delete;patch;watch
+// +kubebuilder:rbac:groups="",resources=serviceaccounts,verbs=get;watch;update;list
+// +kubebuilder:rbac:groups=";apps;extensions",resources=deployments,verbs=create;get;watch;update;delete;list;update;patch
 func (r *ReconcileFunction) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 
-	fnConfigName := "fn-config"
-	fnConfigNamespace := "default"
-	fnConfigNameEnv := os.Getenv("CONTROLLER_CONFIGMAP")
-	fnConfigNamespaceEnv := os.Getenv("CONTROLLER_CONFIGMAP_NS")
+	// Get Function Controller Configuration
+	fnConfig := &corev1.ConfigMap{}
+	if err := r.getFunctionControllerConfiguration(fnConfig); err != nil {
```

Comment:
I think the only reason why is being read is because the RuntimeInfo. We need to know the ServiceAccountName in order to create the Build.

https://github.com/kyma-incubator/runtime/blob/695c0b65d2aa8121404ebbdd89883bcfce06e410/pkg/utils/controller_utils.go#L46

Reply:
Created a follow-up issue for that: https://github.com/kyma-project/kyma/issues/4826